src/modem.rs

```diff
@@ -77,45 +90,69 @@ pub fn wait_for_lte() -> Result<(), Error> {
 	Ok(())
 }
 
+/// Powers the modem on and sets it to auto-register, but does not wait for it
+/// to connect to a network.
+pub fn on() -> Result<(), Error> {
+	crate::at::send_at_command("AT+CFUN=1", |_| {})?;
+	Ok(())
+}
+
+/// Puts the modem into flight mode.
+pub fn flight_mode() -> Result<(), Error> {
+	crate::at::send_at_command("AT+CFUN=4", |_| {})?;
+	Ok(())
+}
+
 /// Powers the modem off.
 pub fn off() -> Result<(), Error> {
-	let skt = crate::at::AtSocket::new()?;
-	// Off
-	skt.write(b"AT+CFUN=0")?;
+	crate::at::send_at_command("AT+CFUN=0", |_| {})?;
 	Ok(())
 }
 
-/// Enable GPS on the nRF9160-DK (PCA10090NS)
+/// Enable GNSS on the nRF9160-DK (PCA10090NS)
 ///
-/// Sends a AT%XMAGPIO command which activates the off-chip GPS RF routing
+/// Sends a AT%XMAGPIO command which activates the off-chip GNSS RF routing
 /// switch when receiving signals between 1574 MHz and 1577 MHz.
 ///
 /// Works on the nRF9160-DK (PCA10090NS) and Actinius Icarus. Other PCBs may
-/// use different MAGPIO pins to control the GPS switch.
-pub fn configure_gps_on_pca10090ns() -> Result<(), Error> {
-	let skt = crate::at::AtSocket::new()?;
-	// Configure the GPS antenna. See `nrf/samples/nrf9160/gps/src/main.c`.
-	skt.write(b"AT%XMAGPIO=1,0,0,1,1,1574,1577")?;
+/// use different MAGPIO pins to control the GNSS switch.
+pub fn configure_gnss_on_pca10090ns() -> Result<(), Error> {
+	// Configure the GNSS antenna. See `nrf/samples/nrf9160/gps/src/main.c`.
+	crate::at::send_at_command("AT%XMAGPIO=1,0,0,1,1,1574,1577", |_| {})?;
 	Ok(())
 }
 
-/// Puts the modem into flight mode.
-pub fn flight_mode() -> Result<(), Error> {
-	let skt = crate::at::AtSocket::new()?;
-	// Flight Mode
-	skt.write(b"AT+CFUN=4")?;
+/// Set which radios should be active. Only works when modem is off.
+pub fn set_system_mode(mode: SystemMode) -> Result<(), Error> {
+	let at_command = match mode {
+		SystemMode::LteM => "AT%XSYSTEMMODE=1,0,0,0",
```

Comment:
It would be better to have the magic configuration strings, 1,0,0,0 | 0,1,0,0 etc defined in one place so that the same strings can be used to set and get the mode.

This would make it easier to change for the next device when the ordering has been changed.

Reply:
Unfortunately the strings aren't the same. To write, we send "AT%XSYSTEMMODE=lte,nbiot,gps,0". That final zero specifies a preference of LTE over NB-IOT (or vice-versa). To read, we look for "%XSYSTEMMODE: lte,nbiot,gps," deliberately leaving off the last zero as we don't care what the preference is.